otter/convergence/transforming.py

```diff
@@ -91,6 +92,30 @@ def optimize_steps(rcv3_steps):
 _register_bulk_rcv3_optimizer(BulkRemoveFromRCv3)
 
 
+def filter_clb_mutating_types(steps):
+    """
+    Allow only one CLB mutating steps per CLB
+
+    :param steps: Iterable of :obj:`IStep` instances
+    :return: Iterable of :obj:`IStep` instances such that only one mutating
+        type is returned per CLB
+    """
+    mutating_clb_types = (AddNodesToCLB, RemoveNodesFromCLB, ChangeCLBNode)
+    lb_step_type = {}
+    for step in steps:
+        stype = type(step)
+        if stype not in mutating_clb_types:
+            yield step
+            continue
+        lb_stype = lb_step_type.get(step.lb_id)
+        if lb_stype is None:
+            lb_step_type[step.lb_id] = stype
```

Comment:
> This algorithm will allow multiple concurrent modifications of the same type, which I think is not what's intended.

This function is called _before_ the steps are combined. Hence it returns multiple steps of the same type. However, things become drastically simple if this function is called _after_ combining the steps whose algorithm is given by you above. I'll make that change. Thanks! 

(How this didn't strike me will hurt me for a long time)